api/src/main/java/org/openmrs/module/appointments/helper/AppointmentServiceHelper.java

```diff
@@ -51,10 +51,14 @@ public String getAppointmentAsJsonString(Appointment appointment) throws IOExcep
         appointmentJson.put("providerUuid", providerUuid);
         String locationUuid = appointment.getLocation() != null ? appointment.getLocation().getUuid() : null;
         appointmentJson.put("locationUuid", locationUuid);
-        appointmentJson.put("startDateTime", appointment.getStartDateTime().toInstant().toString());
-        appointmentJson.put("endDateTime", appointment.getEndDateTime().toInstant().toString());
+        String startDate = (appointment.getStatus() == AppointmentStatus.WaitList && appointment.getStartDateTime() == null) ? null : appointment.getStartDateTime().toInstant().toString();
+        appointmentJson.put("startDateTime", startDate);
+        String endDate = (appointment.getStatus() == AppointmentStatus.WaitList && appointment.getStartDateTime() == null) ? null : appointment.getEndDateTime().toInstant().toString();
```

Comment:
this will negate one of the criterias that a waitlisted appointment can have a date. This assumes that waitlist status appointment can not have a date. Why do we have to do this check against waitlist status. Why not simply just check if the dates are null, alternatively just get the time instant from the date? 

I am assuming the method "getAppointmentAsJsonString" is used for audit purpose, and since it has the appointment object - you have all the info.

Reply:
yeah, this method is used for audit purpose only. This will still allow waitlist appointments with dates by falling to the else part but there is no need of this check against waitlist. I'll remove the status check here